src/proxy/http/router.rs

```diff
@@ -114,27 +115,34 @@ where
 impl<Req, Rec, Mk, B> Stack<Req, Rec, Mk>
 where
     Rec: Recognize<Req> + Clone + Send + Sync + 'static,
+    <Rec as Recognize<Req>>::Target: Send + 'static,
     Mk: rt::Make<Rec::Target> + Clone + Send + Sync + 'static,
-    Mk::Value: svc::Service<Req, Response = http::Response<B>> + Clone,
+    Mk::Value: svc::Service<Req, Response = http::Response<B>> + Clone + Send + 'static,
     <Mk::Value as svc::Service<Req>>::Error: Into<Error>,
     B: Default + Send + 'static,
 {
     pub fn make(&self) -> Service<Req, Rec, Mk> {
-        let inner = Router::new(
+        let (inner, cache_bg) = Router::new(
             self.recognize.clone(),
             self.inner.clone(),
             self.config.capacity,
             self.config.max_idle_age,
         );
+
+        let ctx = logging::Section::Proxy.bg(self.config.proxy_name);
```

Comment:
Nit: this log context will format like this:
`proxy={bg=foo_router}`

I think it might be better to wrap the `proxy_name` field in a type that formats like
```
"cache_purge router={}", proxy_name
```
so that the resulting log lines look like
```
proxy={bg=cache_purge router=foo_router}
```
or something. This might be useful in case there are ever other background contexts associated with routers.

On the other hand, we can always change this later.

Reply:
There is definitely some follow up around this change regarding metrics around the router. I will probably address this then when I do that. These would also be at the `trace` level so I don't think it's too big an issue right now as searching off the `trace` messages is fairly helpful.